pkg/server/sotw/v2/server.go

```diff
@@ -48,15 +48,34 @@ type Callbacks interface {
 	OnStreamResponse(int64, *discovery.DiscoveryRequest, *discovery.DiscoveryResponse)
 }
 
+type Opt func(*server)
+
+func WithResponseBuffer(bufferSize int) Opt {
+	return func(s *server) {
+		s.bufferSize = bufferSize
+	}
+}
+
+func WithCallbacks(callbacks Callbacks) Opt {
+	return func(s *server) {
+		s.callbacks = callbacks
+	}
+}
+
 // NewServer creates handlers from a config watcher and callbacks.
-func NewServer(ctx context.Context, config cache.ConfigWatcher, callbacks Callbacks) Server {
-	return &server{cache: config, callbacks: callbacks, ctx: ctx}
+func NewServer(ctx context.Context, config cache.ConfigWatcher, opts ...Opt) Server {
+	s := &server{cache: config, ctx: ctx, bufferSize: 16}
```

Comment:
can we have a way to override the buffersize?

Reply:
You would basically need to do
`server.NewServerAdvanced(rest.NewServer(...), sotw.NewServer(..., sotw.WithBufferSize(2)))`

Reply:
there is a helper above, working via Opt-s

Reply:
Sorry , i see you have `WithResponseBuffer`